tests/test_graph.py

```diff
@@ -186,7 +186,7 @@ def test_snapshot():
 
 def test_collect():
     x = tensor.matrix()
-    mlp = MLP(activations=[Logistic(), Logistic()], dims=[784, 100, 784],
+    mlp = MLP(activations=[Logistic(), Logistic()], dims=[784, 100, 512],
               use_bias=False)
     cost = SquaredError().apply(x, mlp.apply(x))
```

Comment:
Actually, it should be 784 since we compute MSE between original image and the result of MLP. It doesn't make so big difference in this case, but having 512 may lead to confusion.